site/docs/references/state-parameter-types.md

```diff
@@ -82,32 +86,30 @@ Allows defining a custom state with a **string value**. and then targeting it us
 }
 ```
 
-Setting the state **string value** in the view `<span {...style("root", {selected: "username"})}>` will resolve to `<span data-selected="username" />`.
+Setting the state **string value** in the view `<span {...style("root", {selected: "username"})}>` resolves to `<span data-selected="username" />`.
 
-### Validation [optional]
+### String validation [optional]
 
-You can pass an optional regex string (must be wrapped in quotes) as an argument in order to add validation for the pseudo-class selector **string argument**:
+You can optionally pass a regular expression string as an argument to add validation for the pseudo-class selector **string argument**. The regular expression must be within quotes.
```

Comment:
Missing the fact regex is a validation function that accepts a parameter - maybe something like this?

"You can optionally pass a regular expression function (`regex()`) to the string type to add regular expression validation for the pseudo-class selector. This function accepts a single argument, the expression to be tested against the selector parameter. The expression must be within quotes."